Load a provided CSV dataset into a DataFrame and report its shape, columns, and dtypes:

In [2]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv("train.csv")

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [9]:
print("shape :", df.shape, "\n")
print("columns :",df.columns, "\n")
print("data types :",df.dtypes, "\n")
print(df.info())

shape : (891, 12) 

columns : Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object') 

data types : PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-n

Count missing values per column and handle them (fill or drop) with a documented justification:

In [6]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


justification: we have two columns that contains missing values, I replace missing valeus in Age, and delete column Cabin 'Cause there is too many missing valuse (687/891), it may cause problems in prediction.

In [11]:
df["Age"] = df["Age"].fillna(df["Age"].mean())
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [15]:
df = df.drop(columns=["Cabin"])
print(df.isnull().sum())

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64


Filter the data to a meaningful subset (e.g. rows above a threshold) and describe what you find:

In [21]:
age_mean = df[df["Age"] > 30]
age_mean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 114 entries, 1 to 879
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  114 non-null    int64  
 1   Survived     114 non-null    int64  
 2   Pclass       114 non-null    int64  
 3   Name         114 non-null    object 
 4   Sex          114 non-null    object 
 5   Age          114 non-null    float64
 6   SibSp        114 non-null    int64  
 7   Parch        114 non-null    int64  
 8   Ticket       114 non-null    object 
 9   Fare         114 non-null    float64
 10  Embarked     114 non-null    object 
dtypes: float64(2), int64(5), object(4)
memory usage: 10.7+ KB


there is 114 of 891 Passengers who's age greater than 30 

In [25]:
print(df["Sex"].unique())

['female' 'male']


Use groupby to compute an aggregate statistic per category and interpret the result in a 
Markdown cell:

In [30]:
df.groupby("Pclass").agg({"Age" : "mean", "Fare" : "mean"})

,Age,Fare
Pclass,,
1,36.865551,85.881993
2,25.543695,18.063544
3,22.449853,11.698608


Aggregation describes age and fare avg. of the 3 classes, the avg. age in the 3rd class less than the 1st class, which means that older people tend to travel in high classes and pay more.